### ISSUE TITLE
duplicated_attributes: false positive with apistos crate

### ISSUE THREAD

**Issue by gschulze** [rust-lang/rust-clippy#13238]
### Summary

The `duplicated_attributes` lint yields false positives when using the [apistos](https://github.com/netwo-io/apistos) crate for OpenAPI documentation.

### Lint Name

duplicated_attributes

### Reproducer

I tried this code:

```rust
#[derive(Debug, Serialize, JsonSchema, ApiErrorComponent)]
#[openapi_error(
    status(code = 400, description = "An invalid request has been made."),
    status(code = 404, description = "The specified resource has not been found."),
    status(code = 500, description = "An internal server error occurred.")
)]
pub enum ErrorResponse {
    BadRequest(String),
    InternalServerError(String),
    NotFound(String),
}
```

I saw this happen:

```
warning: duplicated attribute
  --> service/src/api/error_response.rs:12:12
   |
12 |     status(code = 404, description = "The specified resource has not been found."),
   |            ^^^^^^^^^^
   |
note: first defined here
  --> service/src/api/error_response.rs:11:12
   |
11 |     status(code = 400, description = "An invalid request has been made."),
   |            ^^^^^^^^^^
help: remove this attribute
  --> service/src/api/error_response.rs:12:12
   |
12 |     status(code = 404, description = "The specified resource has not been found."),
   |            ^^^^^^^^^^
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#duplicated_attributes
   = note: `-W clippy::duplicated-attributes` implied by `-W clippy::all`
   = help: to override `-W clippy::all` add `#[allow(clippy::duplicated_attributes)]`
```

I expected to see this happen:

No warnings should be emitted.

### Version

```text
rustc 1.80.0 (051478957 2024-07-21)
binary: rustc
commit-hash: 051478957371ee0084a7c0913941d2a8c4757bb9
commit-date: 2024-07-21
host: aarch64-apple-darwin
release: 1.80.0
LLVM version: 18.1.7
```

**Comment by Veetaha** [comment]
Similar flase-positive with the `bon` crate's `#[builder(on(...))]` attribute:

```rust
#[derive(bon::Builder)]
#[builder(on(String, into), on(PathBuf, into))]
pub struct Example {}
```

Here, double `into` is fine according to [`#[builder(on())]`](https://bon-rs.com/reference/builder/top-level/on) semantics.

However, it gets a warning:

```
warning: duplicated attribute
 --> crates/sandbox/src/main.rs:2:41
  |
2 | #[builder(on(String, into), on(PathBuf, into))]
  |                                         ^^^^
  |
note: first defined here
 --> crates/sandbox/src/main.rs:2:22
  |
2 | #[builder(on(String, into), on(PathBuf, into))]
  |                      ^^^^
help: remove this attribute
 --> crates/sandbox/src/main.rs:2:41
  |
2 | #[builder(on(String, into), on(PathBuf, into))]
  |                                         ^^^^
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#duplicated_attributes
  = note: `#[warn(clippy::duplicated_attributes)]` on by default
```

I tried the latest nightly, and I still see a false-positive for `bon`'s attributes

```
cargo 1.88.0-nightly (864f74d4e 2025-04-11)
```

Also mentioned it in this closed issue: https://github.com/rust-lang/rust-clippy/issues/13355#issuecomment-2807773543

**Comment by A-Manning** [comment]
Same issue as #12923

**Comment by InvalidPathException** [comment]
Still present
`src/lib.rs`
```rs
use proc_macro::TokenStream;

#[proc_macro_derive(Demo, attributes(index))]
pub fn demo(_input: TokenStream) -> TokenStream {
    TokenStream::new()
}
```

`tests/repro.rs`
```rs
use clippy_repro::Demo;

#[derive(Demo)]
#[index(a, keys(city))]
#[index(b, keys(city, title))]
pub struct S {
    pub city: String,
    pub title: String,
}
```
